### ISSUE TITLE
how to paint Fig.2 

### ISSUE THREAD

**Issue by SunkiLin** [xjwu1024/PPT#1]
I am particularly interested in painting the scatter plot and distribution plot in Figure 2. They are remarkably well-designed.

**Comment by xjwu1024** [comment]
Thank you for your interest in our work. We ran the inference on the ImageNet validation set with the original DeiT-S model, then stored the **_variance of the significance scores assigned to image tokens at each layer_** into `.pth` files separately. The scatter plot and distribution plot were created using the following code:

```
import torch
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.ticker as ticker

for i in range(12): # 12 blocks
    data = np.empty(shape=(0)) # S_op of each sample
    for j in range(196): # 196 epochs
        data_ = torch.load(f"./{j}_blk_{i}.pth").cpu().numpy()
        data = np.append(data, data_)
    # import pdb; pdb.set_trace()
    print(data.mean())
    index = np.arange(len(data))

    jp = sns.jointplot(x=index, y=data, kind='scatter')  
    jp.ax_marg_x.remove() 
    jp.ax_joint.tick_params(axis='both', labelsize=14)

    # Set x-axis and y-axis tick label format to scientific notation
    plt.gca().yaxis.set_major_formatter(ticker.ScalarFormatter(useMathText=True))
    plt.gca().ticklabel_format(style="sci", axis="y", scilimits=(0,0))

    jp.ax_joint.tick_params(axis='both', which='minor', labelsize=14)

    # Add x-axis and y-axis labels
    plt.xlabel('')
    plt.ylabel('')
    
    plt.savefig(f"./joint_blk_{i}",dpi=150)
    plt.cla()
```

**Comment by SunkiLin** [comment]
Thanks for your prompt help!  Wish you all the best ！
